File: my_py_pkg/my_py_pkg/my_first_node.py
import rclpy # ROS library
from rclpy.node import Node

class MyNode(Node): 
    def __init__(self): 
        super().__init__("py_test")
        self.get_logger().info("Hello ROS2!")


def main(args=None): 
    rclpy.init(args=args) # start ROS2 communication
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__": 
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp" 

class MyNode: public rclcpp::Node
{ 
    public:
        MyNode(): Node("cpp_test")
        {
            RCLCPP_INFO(this->get_logger(), "Hello ROS2 CPP");  
        }

    private: 


};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}